src/binder/expression_binder.cpp

```diff
@@ -111,7 +111,12 @@ static std::string unsupportedImplicitCastException(const Expression& expression
 
 std::shared_ptr<Expression> ExpressionBinder::implicitCastIfNecessary(
     const std::shared_ptr<Expression>& expression, const LogicalType& targetType) {
-    if (expression->dataType == targetType || targetType.containsAny()) { // No need to cast.
+    auto& type = expression->dataType;
+    if ((!type.isInternalType() || !targetType.isInternalType()) &&
```

Comment:
This change (l115-118) is a bit obscure. Let's pull it into a function

```
bool checkImplicitCastWithUDT(type, target) {
if (type.isInternalType() && targetType.isInternalType()) {
      return false;
}
return type.getLogicalTypeID() == targetType.getLogicalTypeID();
}
```

Also give an example in the comment.

You can make the final decision regarding this comment.